byaldi/colpali.py

```diff
@@ -380,7 +380,7 @@ def add_to_index(
         input_item: Union[str, Path, Image.Image, List[Union[str, Path, Image.Image]]],
         store_collection_with_index: bool,
         doc_id: Optional[Union[int, List[int]]] = None,
-        metadata: Optional[List[Dict[str, Union[str, int]]]] = None,
```

Comment:
metadata: Optional[List[Dict[str, Union[str, int]]]] = None 

This should probably be changed to Optional[Union[Dict[str, Union[str, int]], List[Dict[str, Union[str, int]]]]]